packages/ts-interface-generator/README.md

```diff
@@ -0,0 +1,84 @@
+# @ui5/ts-interface-generator
+
+Enables UI5 control development in TypeScript by generating TypeScript interfaces which declare the methods generated by UI5 at runtime.
+
+<b>IMPORTANT: this tool is ALPHA-quality work in progress right now, with plenty of limitations! And there are changes ahead which will likely force users to adapt. Use at your own risk to give feedback. As long as it doesn't crash along the way, it should already be capable of creating nice control interfaces which will save you quite some manual work.<br>
+Note the "TODO" section below as well.<br>
+Also note that this tool does not transpile the TypeScript controls to JavaScript! It only ensures TypeScript development can happen smoothly without being blocked by TypeScript not knowing the methods and structures generated by UI5 at runtime.</b>
```

Comment:
A nice feature for the future would be to reuse the code of the ManagedObject which generates the methods at runtime. Maybe something when going ahead with the ModularCore.

Reply:
That code is written in JavaScript, not TypeScript. ;-) Apart from that, the code is not very isolated, it depends on the ManagedObjectMetadata instance etc.